### ISSUE TITLE
Deletion functionality not working as expected on inline nodes

### ISSUE THREAD

**Issue by TylerRogers94** [ueberdosis/tiptap#3451]
### What’s the bug you are facing?

I've created a custom inline extension which simply renders NodeViewContent and one other component and when I try to delete the node, I am not able to. As demonstrated in the sandbox, if you select all text in the NodeViewContent and then delete, the node size will change to 0, but you can't delete the other element rendered by the extension. If instead you delete 1 character at a time until all the content is gone, you get stuck at node size = 1 and you still cant delete the other element.

Is this a bug or is there something wrong with the way I am rendering my component?

### Which browser was this experienced in? Are any special extensions installed?

Chrome Version 107.0.5304.87

### How can we reproduce the bug on our side?

In CodeSandbox:
- Delete characters from the "Hello World" NodeViewContent until there is nothing left
- Observe that the node size is stuck at 1 and the first element cannot be deleted

### Can you provide a CodeSandbox?

https://codesandbox.io/s/dazzling-james-xn3fnn?file=/src/App.js

### What did you expect to happen?

I expect that the node size would change to 0 when there is no content remaining. I would also expect that when you delete with no content remaining that it would delete the custom extension node

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [X] Yes, I’m a sponsor. 💖

**Comment by stevemckenzie** [comment]
@bdbch any thoughts on this one? The root issue looks to be that the last character doesn't actually get deleted from its state even though you can no longer see it rendered.

**Comment by bdbch** [comment]
Haven't had time to take a look into this. It seems like the whole node view is not rerendered anymore as soon as the last character is deleted. I'll look into it when I find time.

**Comment by stevemckenzie** [comment]
@bdbch yes, I do not see a new render so the node view is stuck with the last character even though it's been deleted.

Thanks for looking into this!

**Comment by theQuazz** [comment]
@bdbch can we get an ETA on this issue? This is a critical part of our application and we're trying to schedule an upcoming release around this bug.

**Comment by bdbch** [comment]
I opened an issue for this in Prosemirror. See this issue:
https://github.com/ProseMirror/prosemirror/issues/1365

Since we won't be able to fix this in a short time right now I'd move it to milestone 2.1.0

**Comment by jordan-sussman** [comment]
For those still facing this issue, you can add a custom backspace handler to detect and force deletion. This works for inline `[node][text][node]` scenarios where you can't delete the last character of text between two inline nodes (in my case, images).
```
  addKeyboardShortcuts() {
    return {
      Backspace: () =>
        this.editor.commands.command(({ tr, state }) => {
          const { $from } = state.selection
          if (
            $from.nodeBefore?.isText &&
            $from.nodeBefore.text?.length === 1 &&
            $from.nodeAfter?.type.name === this.name
          ) {
            tr.delete($from.pos - 1, $from.pos)
            const newPos = tr.mapping.map($from.pos - 1)
            const prevNode = tr.doc.resolve(newPos).nodeBefore
            if (prevNode) {
              tr.setSelection(NodeSelection.create(tr.doc, newPos - prevNode.nodeSize))
            }
            return true
          }
          return false
        }),
    }
  },
```

  The `prevNode` block selects the prior node after deletion. Feel free to omit it if you don't need that behavior. Requires an import of `NodeSelection` from `@tiptap/pm/state`.

**Comment by bdbch** [comment]
Good news. I found a temporary workaround and was able to verify that this problem in fact comes from upstream in combination how we handle keymaps. The problem is that our Backspace handler deletes the selection literally - for a inline text node that would just mean the text. The inline text node will then become invalid but never collapse (that's the actual ProseMirror bug)

I hope this can be fixed upstream so we don't need to do this weird workaround - but now the selection will just be expanded in the scenario that a deletion will remove the whole inline text node.

You can find my upstream report here:
https://code.haverbeke.berlin/prosemirror/prosemirror/issues/1365#issuecomment-69985
